pyproject.toml

```diff
@@ -27,6 +27,7 @@ mkdocs = "^1.1"
 mkdocs-autorefs = "^0.1"
 pymdown-extensions = ">=6.3, <9.0"
 pytkdocs = ">=0.2.0, <0.12.0"
+sphobjinv = "^2.0.1"
```

Comment:
Not an acceptable dependency to always include, in my view. Should be optional.

https://github.com/bskinn/sphobjinv/blob/b8dbe8e975e502bac065519d82a103ea6435aa5b/setup.cfg#L37-L40

Reply:
I think we can even get rid of it. Looking at the uncompressed objects.inv, it seems to be easy to generate:

```
# Sphinx inventory version 2
# Project: project
# Version: 0.0.0
# The remainder of this file is compressed using zlib.
mkdocstrings.extension py:  reference/extension/#mkdocstrings.extension 
mkdocstrings.extension.AutoDocProcessor py:  reference/extension/#mkdocstrings.extension.AutoDocProcessor 
mkdocstrings.extension.AutoDocProcessor.__init__ py:  reference/extension/#mkdocstrings.extension.AutoDocProcessor.__init__ 
mkdocstrings.extension.AutoDocProcessor.run py:  reference/extension/#mkdocstrings.extension.AutoDocProcessor.run 
mkdocstrings.extension.AutoDocProcessor.test py:  reference/extension/#mkdocstrings.extension.AutoDocProcessor.test 
mkdocstrings.extension.MkdocstringsExtension py:  reference/extension/#mkdocstrings.extension.MkdocstringsExtension
```

Reply:
Author of `sphobjinv` here. Yep, it's pretty easy to construct manually. The only really tricky part is the partial zlib compression of only the objects data.

It looks like you're missing a number of elements in the object lines of that inventory; specifically `role`, `priority` and `dispname`. See the syntax for an uncompressed inventory [here](https://sphobjinv.readthedocs.io/en/stable/syntax.html).

Also, FWIW `sphobjinv`'s implementation of the partial zlib compression is [here](https://github.com/bskinn/sphobjinv/blob/260a96cd4eaa7d5cdd8e12f03ae7376243841fa6/src/sphobjinv/zlib.py#L96-L139).

Reply:
Hey @bskinn, thanks a lot for chiming in!

Noted for the zlib partial compression, thanks.

About the elements missing: I guess I can rework the PR so that 1) we don't rely on autorefs to get the registered objects and 2) we also store each objects "type" (class, method), though it depends on the language, so not sure yet how to do this. Then I'll be able to set the role. I can also default the priority to 1. I see that I can also shrink the contents with `$` instead of the name. And add `-` at the end.